Mage/src/main/java/mage/target/targetadjustment/EachOpponentPermanentTargetsAdjuster.java

```diff
@@ -11,36 +11,31 @@
 import java.util.UUID;
 
 /**
- *
  * @author notgreat
  */
-public class EachOpponentPermanentTargetsAdjuster implements TargetAdjuster {
-    private final TargetPermanent blueprintTarget;
+public enum EachOpponentPermanentTargetsAdjuster implements TargetAdjuster {
+    instance;
 
     /**
      * Duplicates the permanent target for each opponent.
      * Filtering of permanent's controllers will be handled inside, so
      * do not pass a blueprint target with a controller restriction filter/predicate.
-     *
-     * @param blueprintTarget The target to be duplicated per opponent
      */
-    public EachOpponentPermanentTargetsAdjuster(TargetPermanent blueprintTarget) {
-        this.blueprintTarget = blueprintTarget.copy(); //Defensively copy the blueprint to ensure immutability
-    }
 
     @Override
     public void adjustTargets(Ability ability, Game game) {
+        TargetPermanent oldTargetPermanent = (TargetPermanent) ability.getTargets().get(0);
```

Comment:
Target data will be lost forever if no one can choose in some point in time. Must use old logic with blueprint. In current use case - can save it on first usage.

![IMG_0168](https://github.com/magefree/mage/assets/8344157/dd916492-ffe0-4039-b9d7-beba3f261abe)

Reply:
Also filter data can be dirty/broken by playerIdPredicate on next abilities usages. So original/blueprint target is important here.

Reply:
Hmm, hidden ability copy all around game engine can help to ignore such problems (target data mutating over time). But I’m not trust to it - it can be changed in the future. Better to write a protective code from a start.

Reply:
Oh, that's why the current `XCMCPermanentAdjuster` only is used for abilities and not for spells, huh. Saving it on the first apply seems like it would work while still keeping the target available to be seen by the triggered ability.

Reply:
XCMCPermanentAdjuster must be used carefully — only with TargetPermanent targets. If something extends target class with new canTarget methods then it will be lost after adjusting (possible bug: allow to select non valid targets). Only TargetPermanent classes or their child with additional predicates compatible with XCMCPermanentAdjuster.

![IMG_0170](https://github.com/magefree/mage/assets/8344157/27e827d5-475c-427e-9b07-1af42340312d)

Reply:
Yeah, I think I'll make a PR soon to convert those adjusters to be more like `EachOpponentPermanentTargetsAdjuster`, copying and modifying a target rather than making a brand new one.